.cmake/check-format.cmake

```diff
@@ -0,0 +1,31 @@
+set (EXCLUDE_DIR "build/")
```

Comment:
If you check the [modern cmake](https://cliutils.gitlab.io/modern-cmake/), they advertise for a `cmake` folder and not `.cmake` folder at the root of the project I would go for that personally ([reference](https://cliutils.gitlab.io/modern-cmake/chapters/basics/structure.html)). Also, try to follow the cmake convention for file name (i.e. CamelCase for modules). So in the end, it should be `cmake/ClangFormat.cmake` or something